docs/advanced-topics/best-practices/best-practices-development.rst

```diff
@@ -0,0 +1,98 @@
+Workflow and Tool Development Process Best Practices
+====================================================
+
+Overview
+--------
+
+This page discusses some best practices in software engineering, and how you can incorporate them into workflow and tool development.
+
+Although the use of software engineering practices in the bioinformatics world is improving, the Dockstore team has noticed that many common best practices, such as automated builds, are not always used in workflow and tool development. This document discusses some ideas and has real-world examples of best practices in bioinformatics.
+
+This is not a complete guide to workflow and tool development, but rather an overview of some concepts to give developers some ideas for features they can incorporate into their development processes. Once you have an automated framework in place, you can keep building on it, adding new features, more tests, etc.
+
+.. note::
+
+   For simplicity's sake, for the rest of this page we'll just use the term *workflow*, instead of saying *workflow and tool*, but all comments apply equally to both tools and workflows.
+
+Automated Builds
+----------------
+
+Automated builds are just that -- builds that trigger automatically. They can by triggered on a schedule (daily, weekly, etc.), by pushes to a branch, by the creation of a pull request or tag, etc.
+
+The advantage of automated builds is that they catch errors right away, before a workflow user spends time and money running the faulty workflow. You can also use automated builds to do security scanning, and if you have a more complex system, package up your builds. 
+
+It's easy to introduce an error to your workflow, from something as simple as missing a closing quotation mark to a logic error in the middle of the program. Having automated builds can quickly catch such problems.
+
+
+Build Servers
+`````````````
+
+There are lots of options for doing automated builds. You can search the web for "Continuous Integration" and get lots of results. As a starting point, here are some services the Dockstore team has used:
+
+- `GitHub Actions <https://docs.github.com/en/actions>`__
+- `CircleCI <https://circle.com>`__
+- `Travis CI <https://www.travis-ci.com>`__
+- `Jenkins <https://www.jenkins.io>`__
+
+GitHub Actions, CircleCI and Travis CI are cloud services; Jenkins is a server you set up in your own infrastructure.
```

Comment:
Maybe throw some "generally" in here. I think there are companies that sell Jenkins as a service and GitHub Actions allows you to farm out jobs to your own infrastructure, but those are exceptions that "generally" probably covers our butt enough